### PR TITLE
Support for Vector Fields for Vector Similarity Search

### PR DESCRIPTION
**Pull request by Avital-Fine**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [x] Does `$ tox` pass with this change (including linting)?
- [x] Do the CI tests pass with this change (enable it first in your forked repo and wait for the github action build to finish)?
- [x] Is the new or changed code fully tested?
- [x] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)?
- [ ] Is there an example added to the examples folder (if applicable)?
- [ ] Was the change added to CHANGES file?

_NOTE: these things are not required to open a PR and can be done
afterwards / while the PR is open._

### Description of change

_Please provide a description of the change here._

### CODE REVIEW

**Review by chayim — APPROVED** [review]

File: tests/test_search.py [hunk 1]
```diff
@@ -1526,6 +1532,27 @@ def test_profile_limited(client):
     assert len(res.docs) == 3  # check also the search result
 
 
+@pytest.mark.redismod
+def test_vector_field(modclient):
+    modclient.flushdb()
+    modclient.ft().create_index(
+        (
+            VectorField(
+                "v", "HNSW", ["TYPE", "FLOAT32", "DIM", 2, "DISTANCE_METRIC", "L2"]
```
Comment (Avital-Fine) [inline tests/test_search.py:1541]: @chayim Maybe it's better to pass the attributes as a dictionary?

Comment (chayim) [reply tests/test_search.py:1541]: It feels a bit clunky. I'd like to two do things before we merge:

1. Try the dictionary approach (and document the kwargs)
2. Add a jupyter notebook example. 

WDYT @Avital-Fine ?

File: redis/commands/search/field.py [hunk 2]
```diff
@@ -90,3 +91,18 @@ def __init__(self, name, separator=",", **kwargs):
...
         )
+
+
+class VectorField(Field):
+    """
+    Allows vector similarity queries against the value in this attribute.
+    See https://oss.redis.com/redisearch/Vectors/#vector_fields.
+    """
+
+    def __init__(self, name, algorithm, attributes, **kwargs):
```
Comment (chayim) [inline redis/commands/search/field.py:102]: + type hints

File: redis/commands/search/field.py [hunk 3]
```diff
@@ -1,15 +1,28 @@
...
+    VECTOR = "VECTOR"
     SORTABLE = "SORTABLE"
     NOINDEX = "NOINDEX"
     AS = "AS"
 
-    def __init__(self, name, args=[], sortable=False, no_index=False, as_name=None):
+    def __init__(
+        self,
+        name: str,
+        args: list[str] = None,
```
Comment (dvora-h) [inline redis/commands/search/field.py:19]: args: List[str] (from typing import List)

File: redis/commands/search/field.py [hunk 4]
```diff
@@ -86,7 +104,54 @@ class TagField(Field):
...
+        if algorithm.upper() not in ["FLAT", "HNSW"]:
+            raise DataError(
+                "Realtime vector indexing supporting 2 Indexing Methods:"
+                "'FLAT' and 'HNSW'."
+            )
+
+        attr_li = []
+
+        for key, value in attributes.items():
+            attr_li.append([key, value])
```
Comment (dvora-h) [inline redis/commands/search/field.py:150]: attr_li.extend([key, value])

**Comment by dvora-h** [comment]
Can you add your example to examples.rst? (and test it locally)

**Comment by dvora-h** [comment]
@Avital-Fine We want to merge it in before 4.2.0 release but the examples were broken so I moved them to a separate branch and after it will be fixed we will merged it too

**Pull request merged**